### ISSUE TITLE
CRC checksum is required for STORED exception when running single-threaded Zip compressor

### ISSUE THREAD

**Issue by Banielisnoob** [Szum123321/textile_backup#130]
When use /backup start,it said An exception occurred when trying to create new backup file

**Comment by tomokp** [comment]
I got the same problem:

`[15:00:29] [pool-3-thread-1/ERROR]: [Textile Backup] An exception occurred when trying to create a new backup file!
java.util.zip.ZipException: CRC checksum is required for STORED method when not writing to a file
        at org.apache.commons.compress.archivers.zip.ZipArchiveOutputStream.validateSizeInformation(ZipArchiveOutputStream.java:943) ~[org_apache_commons_commons-compress-1.22-6f529b8342359d5d.jar:?]
        at org.apache.commons.compress.archivers.zip.ZipArchiveOutputStream.putArchiveEntry(ZipArchiveOutputStream.java:880) ~[org_apache_commons_commons-compress-1.22-6f529b8342359d5d.jar:?]
        at org.apache.commons.compress.archivers.zip.ZipArchiveOutputStream.putArchiveEntry(ZipArchiveOutputStream.java:850) ~[org_apache_commons_commons-compress-1.22-6f529b8342359d5d.jar:?]
        at net.szum123321.textile_backup.core.create.compressors.ZipCompressor.addEntry(ZipCompressor.java:77) ~[textile_backup-3.1.0-1.20.jar:?]
        at net.szum123321.textile_backup.core.create.compressors.AbstractCompressor.createArchive(AbstractCompressor.java:99) ~[textile_backup-3.1.0-1.20.jar:?]
        at net.szum123321.textile_backup.core.create.ExecutableBackup.call(ExecutableBackup.java:107) ~[textile_backup-3.1.0-1.20.jar:?]
        at net.szum123321.textile_backup.core.create.ExecutableBackup.call(ExecutableBackup.java:25) ~[textile_backup-3.1.0-1.20.jar:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
        Suppressed: java.io.IOException: This archive contains unclosed entries.
                at org.apache.commons.compress.archivers.zip.ZipArchiveOutputStream.finish(ZipArchiveOutputStream.java:563) ~[org_apache_commons_commons-compress-1.22-6f529b8342359d5d.jar:?]
                at org.apache.commons.compress.archivers.zip.ZipArchiveOutputStream.close(ZipArchiveOutputStream.java:1119) ~[org_apache_commons_commons-compress-1.22-6f529b8342359d5d.jar:?]
                at net.szum123321.textile_backup.core.create.compressors.AbstractCompressor.createArchive(AbstractCompressor.java:52) ~[textile_backup-3.1.0-1.20.jar:?]
                at net.szum123321.textile_backup.core.create.ExecutableBackup.call(ExecutableBackup.java:107) ~[textile_backup-3.1.0-1.20.jar:?]
                at net.szum123321.textile_backup.core.create.ExecutableBackup.call(ExecutableBackup.java:25) ~[textile_backup-3.1.0-1.20.jar:?]
                at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
                at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
                at java.lang.Thread.run(Thread.java:833) ~[?:?]`

**Comment by Kongtu5i** [comment]
i have the same question,my minecraft version is 1.20.1, java version is 17.0.2

**Comment by iewnfod** [comment]
Save Problem

Error Message:
```
[07:40:29] [pool-5-thread-1/ERROR]: [Textile Backup] An exception occurred when trying to create a new backup file!
java.util.zip.ZipException: CRC checksum is required for STORED method when not writing to a file
        at org.apache.commons.compress.archivers.zip.ZipArchiveOutputStream.validateSizeInformation(ZipArchiveOutputStream.java:943) ~[org_apache_commons_commons-compress-1.22-6f529b8342359d5d.jar:?]
        at org.apache.commons.compress.archivers.zip.ZipArchiveOutputStream.putArchiveEntry(ZipArchiveOutputStream.java:880) ~[org_apache_commons_commons-compress-1.22-6f529b8342359d5d.jar:?]
        at org.apache.commons.compress.archivers.zip.ZipArchiveOutputStream.putArchiveEntry(ZipArchiveOutputStream.java:850) ~[org_apache_commons_commons-compress-1.22-6f529b8342359d5d.jar:?]
        at net.szum123321.textile_backup.core.create.compressors.ZipCompressor.addEntry(ZipCompressor.java:77) ~[textile_backup-3.1.0-1.20.jar:?]
        at net.szum123321.textile_backup.core.create.compressors.AbstractCompressor.createArchive(AbstractCompressor.java:99) ~[textile_backup-3.1.0-1.20.jar:?]
        at net.szum123321.textile_backup.core.create.ExecutableBackup.call(ExecutableBackup.java:107) ~[textile_backup-3.1.0-1.20.jar:?]
        at net.szum123321.textile_backup.core.create.ExecutableBackup.call(ExecutableBackup.java:25) ~[textile_backup-3.1.0-1.20.jar:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
        Suppressed: java.io.IOException: This archive contains unclosed entries.
                at org.apache.commons.compress.archivers.zip.ZipArchiveOutputStream.finish(ZipArchiveOutputStream.java:563) ~[org_apache_commons_commons-compress-1.22-6f529b8342359d5d.jar:?]
                at org.apache.commons.compress.archivers.zip.ZipArchiveOutputStream.close(ZipArchiveOutputStream.java:1119) ~[org_apache_commons_commons-compress-1.22-6f529b8342359d5d.jar:?]
                at net.szum123321.textile_backup.core.create.compressors.AbstractCompressor.createArchive(AbstractCompressor.java:52) ~[textile_backup-3.1.0-1.20.jar:?]
                at net.szum123321.textile_backup.core.create.ExecutableBackup.call(ExecutableBackup.java:107) ~[textile_backup-3.1.0-1.20.jar:?]
                at net.szum123321.textile_backup.core.create.ExecutableBackup.call(ExecutableBackup.java:25) ~[textile_backup-3.1.0-1.20.jar:?]
                at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
                at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
                at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

Minecraft Version: 1.20.1
Java Version: OpenJDK 17
Fabric Loader Version: 0.14.21

**Comment by Szum123321** [comment]
Issue appears in TxB > 3.0.0 when running single-threaded zip compressor [ZipCompressor.java](https://github.com/Szum123321/textile_backup/blob/2.x/src/main/java/net/szum123321/textile_backup/core/create/compressors/ZipCompressor.java) .

**Comment by Szum123321** [comment]
I am working on a solution

**Comment by Szum123321** [comment]
fixed in 3.1.2
